### ISSUE TITLE
`.failing` isn't enclosed in #if DEBUG thus breaks release builds

### ISSUE THREAD

**Issue by andreyz** [pointfreeco/composable-core-location#14]
Enclosing extension in `#if DEBUG` `#endif` will solve it.
https://github.com/pointfreeco/composable-core-location/blob/95c32ab3677023aed0f5eb18c5c312bc31e296cb/Sources/ComposableCoreLocation/Failing.swift#L5

**Comment by andreyz** [comment]
Ehh! Just realised that we've misunderstood each other and the PR we've merged didn't help fix package building for Release.

The reason why this package fails to build for release is that Effect.failing is only defined for Debug in TCA, while our earlier discussion was around XCTFail (which is indeed stubbed for Release in XCTestDynamicOverlay).

I'm ready to make another PR. What's best plan of action:

Implement stubbed Effect.failing in ComposableCoreLocation or
Enclose LocationManager.failing in #if DEBUG/#endif?

**Comment by andreyz** [comment]
Maybe even better solution is to remove #if DEBUG/#endif from TCA's definition of [`Effect.failing`](https://github.com/pointfreeco/swift-composable-architecture/blob/03a814d0d35474efdc5114cbcd8db4965758587a/Sources/ComposableArchitecture/TestSupport/FailingEffect.swift#L85)?

**Comment by mbrandonw** [comment]
> Maybe even better solution is to remove #if DEBUG/#endif from TCA's definition of [`Effect.failing`](https://github.com/pointfreeco/swift-composable-architecture/blob/03a814d0d35474efdc5114cbcd8db4965758587a/Sources/ComposableArchitecture/TestSupport/FailingEffect.swift#L85)?

Yes, let's do that! These `#if DEBUG`s are a mess and I think this one is just left over from before we had `XCTFail` defined for release builds.

**Comment by andreyz** [comment]
OK! Then I'll make a PR on TCA repo. Once that get's merged, I can make a follow up PR here to pin this package to TCA revision of version containing the #if DEBUG/#endif removal from definition of `Effect.failing`.

**Comment by mbrandonw** [comment]
Sounds good! 

Also if you need to be unblocked immediately I suppose you could wrap your failing effects in `#if DEBUG` in the meantime.

**Comment by stephencelis** [comment]
@andreyz We just released [swift-composable-architecture 0.27.1](https://github.com/pointfreeco/swift-composable-architecture/releases/0.27.1), so feel free to submit that PR. We can also do it. Just wanted to give you an opportunity to first 😄

**Comment by andreyz** [comment]
Yeah! I'll only have time available in 5-6 hours, feel free to do it before.

**Comment by andreyz** [comment]
Updated the PR just now.
